applications/FluidDynamicsApplication/custom_processes/shock_capturing_entropy_viscosity_process.cpp

```diff
@@ -0,0 +1,390 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main author:     Eduard Gómez
+//
+
+#include <cmath>
+#include <tuple>
+
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+#include "processes/calculate_nodal_area_process.h"
+
+#include "fluid_dynamics_application_variables.h"
+#include "shock_capturing_entropy_viscosity_process.h"
+
+
+namespace Kratos {
+
+
+ShockCapturingEntropyViscosityProcess::ShockCapturingEntropyViscosityProcess(
+    ModelPart& rModelPart,
+    Parameters rParameters)
+    : Process()
+    , mrModelPart(rModelPart)
+{
+    KRATOS_TRY
+
+    rParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+
+    mComputeAreasEveryStep = rParameters["calculate_nodal_area_at_each_step"].GetBool();
+    mEntropyConstant = rParameters["entropy_constant"].GetDouble();     // In the article: c_e
+    mEnergyConstant = rParameters["energy_constant"].GetDouble();   // In the article: c_max
+
+    mArtificialMassDiffusivityPrandtl = rParameters["artificial_mass_viscosity_Prandtl"].GetDouble();
+    mArtificialConductivityPrandtl = rParameters["artificial_conductivity_Prandtl"].GetDouble();
+
+    mArtificialMassDiffusivityPrandtl = 0.0;
+
+
+    KRATOS_CATCH("")
+}
+
+
+int ShockCapturingEntropyViscosityProcess::Check()
+{
+    const auto err_code = Process::Check();
```

Comment:
```suggestion
    const int err_code = Process::Check();
```
(it is common practice to avoid `auto` for simple known types).